master/pkg/tasks/task.go

```diff
@@ -124,6 +130,14 @@ func (t *TaskSpec) ToDockerSpec() container.Spec {
 	}
 	envVars = append(envVars, env.EnvironmentVariables().For(deviceType)...)
 
+	containerRuntime := ""
+	switch deviceType {
+	case device.CPU, device.ZeroSlot:
+		containerRuntime = runc
+	case device.GPU:
+		containerRuntime = nvidiaContainerRuntime
+	}
```

Comment:
safety default?

```
default:
    panic(fmt.Sprintf("bad device type: %s", deviceType))
```

Reply:
for now it should just run the default docker container runtime if there is an unrecognized device, which i think is fine. anyway, i think we'll panic elsewhere if it was invalid.

Reply:
eh supposed it cant hurt.

Reply:
I remember using some [functional] programming language where one 
1. had to make a string enum 
2. do a switch on that strongly typed enum (instead of raw const/literal strings) 
3. compiler would complain if you didn't have a `case` for each possible type value, or a default. 

it was opinionated but helpful chunk of static analysis.

Reply:
yeah, i definitely like that. i think go help some linter for that (found it https://github.com/nishanths/exhaustive)